frontend/src/lib/components/ActivityLog/activityLogLogic.tsx

```diff
@@ -1,30 +1,100 @@
             {
-                fetchActivity: async () => {
-                    const url = props.id
-                        ? `/api/projects/@current/feature_flags/${props.id}/activity`
-                        : `/api/projects/@current/feature_flags/activity`
-                    const apiResponse: PaginatedResponse<ActivityLogItem> = await api.get(url)
-                    return humanize(apiResponse?.results, props.describer)
+                fetchPreviousPage: async () => {
+                    const url = values.previousPageURL
+                    return url === null ? null : await api.get(url)
                 },
             },
         ],
     }),
+    reducers: ({ props }) => ({
+        page: [
+            props.startingPage ? props.startingPage - 1 : 0,
+            {
+                fetchNextPageSuccess: (state) => state + 1,
+                fetchPreviousPageSuccess: (state) => state - 1,
+            },
+        ],
+        humanizedActivity: [
+            [] as HumanizedActivityLogItem[],
+            {
+                fetchNextPageSuccess: (state, { nextPage }) =>
+                    nextPage ? humanize(nextPage.results, props.describer) : state,
+                fetchPreviousPageSuccess: (state, { previousPage }) =>
+                    previousPage ? humanize(previousPage.results, props.describer) : state,
+            },
+        ],
+        previousPageURL: [
+            null as string | null,
+            {
+                fetchNextPageSuccess: (_, { nextPage }) => nextPage.previous || null,
+                fetchPreviousPageSuccess: (_, { previousPage }) => previousPage.previous || null,
+            },
+        ],
+        totalCount: [
+            null as number | null,
+            {
+                fetchNextPageSuccess: (_, { nextPage }) => nextPage.total_count || null,
+                fetchPreviousPageSuccess: (_, { previousPage }) => previousPage.total_count || null,
+            },
+        ],
+        nextPageURL: [
+            (props.id
+                ? `/api/projects/@current/feature_flags/${props.id}/activity?page=${props.startingPage || 1}`
+                : `/api/projects/@current/feature_flags/activity?page=${props.startingPage || 1}`) as string | null,
+            {
+                fetchNextPageSuccess: (_, { nextPage }) => nextPage.next || null,
+                fetchPreviousPageSuccess: (_, { previousPage }) => previousPage.next || null,
+            },
+        ],
+    }),
+    selectors: ({ actions }) => ({
+        hasNextPage: [(s) => [s.nextPageURL], (nextPageURL: string | null) => !!nextPageURL],
+        hasPreviousPage: [(s) => [s.previousPageURL], (previousPageURL: string | null) => !!previousPageURL],
+        pagination: [
+            (s) => [s.page, s.totalCount],
+            (page, totalCount): PaginationManual => {
+                return {
+                    controlled: true,
+                    pageSize: 10,
```

Comment:
10 seems very small when using it, as it even leaves a lot of whitespace on a laptop screen. How about 20 or30?

Reply:
My laptop resolution only shows 8... Time for an eye test for me maybe 👀 

Set to 20. It's an easy change...